src/main/java/com/heeverse/member/service/MemberService.java

```diff
@@ -1,8 +1,38 @@
 package com.heeverse.member.service;
 
+import com.heeverse.member.domain.entity.Member;
+import com.heeverse.member.domain.repository.MemberRepository;
+import com.heeverse.member.dto.MemberRequestDto;
+import com.heeverse.member.exception.DuplicatedMemberException;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class MemberService {
 
+    private final PasswordEncoder passwordEncoder;
+    private final MemberRepository memberRepository;
+
+    public MemberService(PasswordEncoder passwordEncoder, MemberRepository memberRepository) {
+        this.passwordEncoder = passwordEncoder;
+        this.memberRepository = memberRepository;
+    }
+
+    public void signup(MemberRequestDto memberRequestDto) {
+        boolean existMember = isExistMember(memberRequestDto);
+        if (existMember) {
+            throw new DuplicatedMemberException();
+        }
```

Comment:
코드 블럭 내부에 변수가 많을수록 코드를 읽기 어려워집니다.
이렇게 조건으로만 사용하는 변수의 경우 짧게 줄여주시면 불필요한 변수 선언을 줄일 수 있습니다. 코드 읽기도 더 쉬워져요~
```suggestion
        if (isExistMember(memberRequestDto)) {
            throw new DuplicatedMemberException();
        }
```